MoltenVK/MoltenVK/GPUObjects/MVKPixelFormats.mm

```diff
@@ -1313,25 +1294,23 @@
 	addMTLPixelFormatDesc    ( X24_Stencil8, Depth24_Stencil8, None, DRMR );
 	addMTLPixelFormatDesc    ( X32_Stencil8, Depth32_Stencil8, DRM, DRMR );
 
-	// When adding to this list, be sure to ensure _mtlPixelFormatCount is large enough for the format count
+	_mtlPixelFormatDescriptions.shrink_to_fit();
 }
 
+// If necessary, resize vector with empty elements
 #define addMTLVertexFormatDesc(MTL_VTX_FMT, IOS_CAPS, MACOS_CAPS)  \
-	MVKAssert(fmtIdx < _mtlVertexFormatCount, "Attempting to describe %d MTLVertexFormats, but only have space for %d. Increase the value of _mtlVertexFormatCount", fmtIdx + 1, _mtlVertexFormatCount);  \
-	_mtlVertexFormatDescriptions[fmtIdx++] = { .mtlVertexFormat = MTLVertexFormat ##MTL_VTX_FMT, VK_FORMAT_UNDEFINED,  \
-                                               mvkSelectPlatformValue<MVKMTLFmtCaps>(kMVKMTLFmtCaps ##MACOS_CAPS, kMVKMTLFmtCaps ##IOS_CAPS),  \
-                                               MVKMTLViewClass::None, MTLPixelFormatInvalid, "MTLVertexFormat" #MTL_VTX_FMT }
+	mtlVtxFmt = MTLVertexFormat ##MTL_VTX_FMT;  \
+if (mtlVtxFmt >= _mtlVertexFormatDescriptions.size()) { _mtlVertexFormatDescriptions.resize(mtlVtxFmt + 1, {}); }  \
+	_mtlVertexFormatDescriptions[mtlVtxFmt] = { .mtlVertexFormat = mtlVtxFmt, VK_FORMAT_UNDEFINED,  \
+												mvkSelectPlatformValue<MVKMTLFmtCaps>(kMVKMTLFmtCaps ##MACOS_CAPS, kMVKMTLFmtCaps ##IOS_CAPS),  \
+												MVKMTLViewClass::None, MTLPixelFormatInvalid, "MTLVertexFormat" #MTL_VTX_FMT };
```

Comment:
`initMTLVertexFormatCapabilities` calls `reserve` on the vector initially with `MTLVertexFormatHalf + 3` and `resize` will be called for every new format description. It won't re-allocate, as reserve and resize never make the allocation smaller, but this is still bad usage of resize, as its being called many times for effectively no reason.

We should probably just use `resize` at the start of the function to drastically reduce the number of calls to resize made in this function. 

Also, is there any reason why we use designated initializers for the `mtlVertexFormat` field here, but not for any other fields of the struct?

Reply:
> We should probably just use resize at the start of the function to drastically reduce the number of calls to resize made in this function.

Good point. I've changed `reserve()` to `resize()`. The `resize()` in `addMTLVertexFormatDesc()` will then only be engaged if we call that function more times than the initial `resize()` reserves.

> Also, is there any reason why we use designated initializers for the mtlVertexFormat field here, but not for any other fields of the struct?

Yes. Because that field is a union, the compiler doesn't know how to build the initializer list otherwise.